### PR TITLE
Blogroll : Block refactor, styling changes, and additional enhancements

### PR DESCRIPTION
**Pull request by agrullon95**
Fixes:
https://github.com/Automattic/wp-calypso/issues/81351

## Proposed changes:

Fix default icon image
Fix icon styling for width and height
Remove Blogroll settings and setting context values

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to  the editor
* Add the blogroll block
* Ensure that the response from `recommendations` is used to create the initial content.
* Ensure that the suggestions are returning correctly.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/blogroll/edit.js [hunk 1]
```diff
@@ -37,8 +39,20 @@ export function BlogRollEdit( { className, attributes, setAttributes, clientId }
...
+		setAdditionalClasses( [
+			...( ! show_avatar ? [ 'hide-avatar' ] : [] ),
+			...( ! show_description ? [ 'hide-description' ] : [] ),
+			...( ! show_subscribe_button ? [ 'hide-subscribe-button' ] : [] ),
+		] );
+	}, [ show_avatar, show_description, show_subscribe_button ] );
+
+	const blockProps = useBlockProps( {
+		className: classNames( className, ...additionalClasses ),
+	} );
```
Comment (alshakero) [inline projects/plugins/jetpack/extensions/blocks/blogroll/edit.js]: No need to store additional classes in a state. They're derivative.

```suggestion
	const blockProps = useBlockProps( {
		className: classNames( className, {
			'hide-avatar': ! show_avatar,
			'hide-description': ! show_description,
			'hide-subscribe-button': ! show_subscribe_button
		} ),
	} );
```

File: projects/plugins/jetpack/extensions/blocks/blogroll/index.js [hunk 2]
```diff
@@ -61,7 +58,24 @@ export const settings = {
...
+	save: ( { attributes: { show_avatar, show_description, show_subscribe_button }, className } ) => {
+		const additionalClasses = [
+			...( ! show_avatar ? [ 'hide-avatar' ] : [] ),
+			...( ! show_description ? [ 'hide-description' ] : [] ),
+			...( ! show_subscribe_button ? [ 'hide-subscribe-button' ] : [] ),
+		];
+
+		const blockProps = useBlockProps.save( {
+			className: classNames( className, ...additionalClasses ),
+		} );
```
Comment (alshakero) [inline projects/plugins/jetpack/extensions/blocks/blogroll/index.js]: +1

```suggestion
		const blockProps = useBlockProps.save( {
			className: classNames( className, {
				'hide-avatar': ! show_avatar,
				'hide-description': ! show_description,
				'hide-subscribe-button': ! show_subscribe_button
			} ),
		} ); );
```

File: projects/plugins/jetpack/extensions/blocks/blogroll/index.js [hunk 3]
```diff
@@ -61,7 +58,24 @@ export const settings = {
...
+
+		return (
+			<div { ...blockProps }>
+				{ ' ' }
+				<InnerBlocks.Content />{ ' ' }
+			</div>
```
Comment (alshakero) [inline projects/plugins/jetpack/extensions/blocks/blogroll/index.js]: Do we need these white spaces?

**Review by alshakero — CHANGES_REQUESTED** [review]
Looks great. Left some enhancements.

File: projects/plugins/jetpack/extensions/blocks/blogroll/edit.js (hunk 1 above)
Comment (heavyweight) [reply projects/plugins/jetpack/extensions/blocks/blogroll/edit.js]: agh yes fixing this

File: projects/plugins/jetpack/extensions/blocks/blogroll/index.js (hunk 3 above)
Comment (heavyweight) [reply projects/plugins/jetpack/extensions/blocks/blogroll/index.js]: Whoa. For sure no. This probably happened by mistake

**Review by alshakero — CHANGES_REQUESTED** [review]
Looks great! Please fix conflicts so I can test.

File: projects/plugins/jetpack/extensions/blocks/blogroll/block.json [hunk 4]
```diff
@@ -63,5 +56,11 @@
 			"type": "boolean",
 			"default": true
 		}
+	},
+	"providesContext": {
+		"showAvatar": "show_avatar",
+		"showDescription": "show_description",
+		"showSubscribeButton": "show_subscribe_button",
+		"openLinksNewWindow": "open_links_new_window"
```
Comment (alshakero) [inline projects/plugins/jetpack/extensions/blocks/blogroll/block.json:64]: If I'm not mistaken, only two are used

```suggestion
		"showSubscribeButton": "show_subscribe_button",
		"openLinksNewWindow": "open_links_new_window"
```

File: projects/plugins/jetpack/extensions/blocks/blogroll/style.scss [hunk 5]
```diff
@@ -2,4 +2,37 @@
...
+		font-weight: 400;
+		color: var(--gray-gray-50, #646970);
+
+		.jetpack-blogroll-item-title {
+			color: var(--gray-gray-100, #101517);
+			font-size: 16px;
+			font-weight: 500;
+			margin: 0;
+			line-height: 24px;
+		}
```
Comment (alshakero) [inline projects/plugins/jetpack/extensions/blocks/blogroll/style.scss:21]: Shouldn't the color and the font size be customizable?

File: projects/plugins/jetpack/extensions/blocks/blogroll/use-recommendations.js [hunk 6]
```diff
@@ -2,13 +2,17 @@ import apiFetch from '@wordpress/api-fetch';
...
-export default function useRecommendations() {
+export default function useRecommendations( enabled = false ) {
 	const [ isLoading, setIsLoading ] = useState( false );
 	const [ errorMessage, setErrorMessage ] = useState( null );
 	const [ recommendations, setRecommendations ] = useState( [] );
 	const abortControllerRef = useRef();
 
 	useEffect( () => {
+		if ( ! enabled ) {
+			return;
```
Comment (alshakero) [inline projects/plugins/jetpack/extensions/blocks/blogroll/use-recommendations.js:14]: let's set `isLoading` false here. And we should pass `enabled` to deps below.

File: projects/plugins/jetpack/extensions/blocks/blogroll/utils.js [hunk 7]
```diff
@@ -5,21 +5,10 @@ export function createBlockFromRecommendation( attrs ) {
 
 	return createBlock( 'jetpack/blogroll-item', {
 		...attrs,
-		icon: getSiteIconOrPlaceholder( icon ),
+		icon: icon || 'https://s0.wp.com/i/webclip.png',
```
Comment (alshakero) [inline projects/plugins/jetpack/extensions/blocks/blogroll/utils.js:8]: No longer needed.

**Review by alshakero — CHANGES_REQUESTED** [review]
Looove it. Left some tiny comments.

File: projects/plugins/jetpack/extensions/blocks/blogroll/utils.js (hunk 7 above)
Comment (agrullon95) [reply projects/plugins/jetpack/extensions/blocks/blogroll/utils.js:8]: We should keep this to display a placeholder image in the editor.
If I remove this and the appended site doesn't have an icon, the image is broken in the editor.

File: projects/plugins/jetpack/extensions/blocks/blogroll/use-recommendations.js (hunk 6 above)
Comment (agrullon95) [reply projects/plugins/jetpack/extensions/blocks/blogroll/use-recommendations.js:14]: Updated

File: projects/plugins/jetpack/extensions/blocks/blogroll/blogroll-item/blogroll-item.php [hunk 8]
```diff
@@ -23,7 +23,10 @@
 function register_block() {
```
Comment (agrullon95) [inline projects/plugins/jetpack/extensions/blocks/blogroll/blogroll-item/blogroll-item.php:23]: Test changes for customizable color/font-size:

```
function load_assets( $attr, $content, $block ) {
	/*
	 * Enqueue necessary scripts and styles.
	 */
	Jetpack_Gutenberg::load_assets_as_required( FEATURE_NAME );

	$default_title_font_size   = 16;
	$default_title_color       = '#101517';
	$default_description_color = '#646970';

	$name              = esc_attr( $attr['name'] );
	$url               = esc_url( $attr['url'] );
	$description       = esc_attr( $attr['description'] );
	$icon              = esc_attr( $attr['icon'] );
	$title_font_size   = esc_attr( isset( $attr['titleFontSize'] ) ? $attr['titleFontSize'] : $default_title_font_size );
	$title_color       = esc_attr( isset( $attr['titleColor'] ) ? $attr['titleColor'] : $default_title_color );
	$description_color = esc_attr( isset( $attr['descriptionColor'] ) ? $attr['descriptionColor'] : $default_description_color );
	$background_color  = esc_attr( isset( $attr['backgroundColor'] ) ? $attr['backgroundColor'] : '#ffffff' );

	$target = $block->context['openLinksNewWindow'] ? '_blank' : '_self';

	if ( empty( $icon ) ) {
		$icon = 'https://s0.wp.com/i/webclip.png';
	}

	$content = <<<HTML
<figure>
	<img src="$icon" alt="$name">
</figure>
<div class="test">
	<a class="jetpack-blogroll-item-title" href="$url" target="$target" rel="noopener noreferrer" style="color:$title_color; font-size:$title_font_size">$name</a>
	<div class="jetpack-blogroll-item-description" style="color:$description_color">$description</div>
</div>
HTML;

	return sprintf(
		'<div style="background-color:$class="%1$s">%2$s</div>',
		esc_attr( Blocks::classes( FEATURE_NAME, $attr ) ),
		$content
	);
}
```

File: projects/plugins/jetpack/extensions/blocks/blogroll/blogroll-item/edit.js [hunk 9]
```diff
@@ -1,68 +1,49 @@
-import { InnerBlocks } from '@wordpress/block-editor';
+import { RichText, MediaUpload, useBlockProps } from '@wordpress/block-editor';
```
Comment (agrullon95) [inline projects/plugins/jetpack/extensions/blocks/blogroll/blogroll-item/edit.js:1]: Test changes for customizable color/font size:

```
function BlogrollItemEdit( { className, attributes, setAttributes } ) {
	const { icon, name, description, titleFontSize, titleColor, backgroundColor, descriptionColor } =
		attributes;
	const blockProps = useBlockProps( { className } );

	return (
		<div { ...blockProps } style={ { backgroundColor: backgroundColor } }>
			<MediaUpload
				multiple={ false }
				onSelect={ media => {
					setAttributes( { icon: media.url } );
				} }
				render={ ( { open } ) => (
					<Button variant="link" onClick={ open } style={ { padding: 0 } }>
						<figure>
							<img src={ icon } alt={ name } />
						</figure>
					</Button>
				) }
			/>
			<div>
				<a>
					<InspectorControls>
						<PanelColorSettings
							title={ __( 'Color Settings', 'jetpack' ) }
							colorSettings={ [
								{
									value: titleColor,
									onChange: value => setAttributes( { titleColor: value } ),
									label: __( 'Text', 'jetpack' ),
								},
								{
									value: backgroundColor,
									onChange: value => setAttributes( { backgroundColor: value } ),
									label: __( 'Background', 'jetpack' ),
								},
								{
									value: descriptionColor,
									onChange: value => setAttributes( { descriptionColor: value } ),
									label: __( 'Description', 'jetpack' ),
								},
							] }
						/>
						<PanelBody title={ __( 'Typography', 'jetpack' ) }>
							<FontSizePicker
								value={ titleFontSize }
								fallbackFontSize={ 16 }
								onChange={ value => setAttributes( { titleFontSize: value } ) }
								disableCustomFontSizes={ true }
							/>
						</PanelBody>
					</InspectorControls>
					<RichText
						style={ {
							fontSize: titleFontSize,
							color: titleColor,
						} }
						className="jetpack-blogroll-item-title"
						value={ name }
						tagName={ 'h3' }
						allowedFormats={ [ 'core/bold', 'core/italic' ] }
						onChange={ value => {
							setAttributes( { name: value } );
						} }
						placeholder={ __( 'Enter site title', 'jetpack' ) }
					/>
				</a>
				<RichText
					style={ {
						color: descriptionColor,
					} }
					className="jetpack-blogroll-item-description"
					value={ description }
					onChange={ value => {
						setAttributes( { description: value } );
					} }
					placeholder={ __( 'Enter site description', 'jetpack' ) }
				/>
			</div>
		</div>
	);
}
```

File: projects/plugins/jetpack/extensions/blocks/blogroll/blogroll-item/index.js [hunk 10]
```diff
@@ -47,9 +42,25 @@ export const settings = {
...
+			type: 'string',
+			default: 'https://s0.wp.com/i/webclip.png',
+		},
+		url: {
+			type: 'string',
+		},
+		description: {
+			type: 'string',
+		},
+	},
```
Comment (agrullon95) [inline projects/plugins/jetpack/extensions/blocks/blogroll/blogroll-item/index.js:62]: Test changes for customizable color/font-sizes:

```
attributes: {
		id: {
			type: 'string',
		},
		name: {
			type: 'string',
		},
		icon: {
			type: 'string',
			default: 'https://s0.wp.com/i/webclip.png',
		},
		url: {
			type: 'string',
		},
		description: {
			type: 'string',
		},
		titleFontSize: {
			type: 'number',
			default: 16,
		},
		descriptionFontSize: {
			type: 'number',
			default: 14,
		},
		titleColor: {
			type: 'string',
			default: '#101517',
		},
		backgroundColor: {
			type: 'string',
		},
		descriptionColor: {
			type: 'string',
			default: '#646970',
		},
	},
```

**Review by alshakero — APPROVED** [review]
Looks great. Let's merge this and do a final clean up PR. I'm confused by this many PRs.

**Pull request merged**